Makefile

```diff
@@ -44,21 +48,30 @@ setup:
 get_thrift:
 	scripts/travis/get-thrift.sh
 
+# We want to remove `vendor` dir because thrift-gen tests don't work with it.
+# However, glide install even with --cache-gopath option leaves GOPATH at HEAD,
+# not at the desired versions from glide.lock, which are only applied to `vendor`
+# dir. So we move `vendor` to a temp dir and prepend it to GOPATH.
+# Note that glide itself is still executed against the original GOPATH.
 install:
-	glide --debug install --cache --cache-gopath
-	rm -rf vendor
+	GOPATH=$(OLD_GOPATH) glide --debug install --cache --cache-gopath
+	rm -rf $(VENDOR_PATH)
+	mkdir -p $(VENDOR_PATH)/src
```

Comment:
you probably shouldn't create `src` -- otherwise the `mv` below might move to `$(VENDOR_PATH)/src/vendor`. easier to just `mkdir -p $(VENDOR_PATH)`